framework/Util/TEscCharsetConverter.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * TEscCharsetConverter class file
+ *
+ * @author Brad Anderson <belisoful@icloud.com>
+ * @link https://github.com/pradosoft/prado
+ * @license https://github.com/pradosoft/prado/blob/master/LICENSE
+ */
+
+namespace Prado\Util;
+
+/**
+ * TEscCharsetConverter class.
+ *
+ * TEscCharsetConverter is the ESC Charset Converter for converting between ESC
+ * character sets] encodings and their iConv character encodings.
+ *
+ * Each Esc charset Encoding has 4 versions for G0, G1, G2, and G3.
+ *
+ * @author Brad Anderson <belisoful@icloud.com>
+ * @since 4.2.3
+ * @see https://en.wikipedia.org/wiki/ISO/IEC_2022 General structure of character encodings.
+ * @see https://en.wikipedia.org/wiki/ISO/IEC_646 National standards for ASCII.
+ * @see https://www.sljfaq.org/afaq/encodings.html Japanese encodings and character sets.
+ *
+ * These are not yet in iconv (as of April, 2023):
+ * @todo missing ISO-5427. https://en.wikipedia.org/wiki/ISO_5427. (8 bit Cyrillic, 1979/1981)
+ * @todo missing ČSN (Czech technical standard) 369103. https://en.wikipedia.org/wiki/KOI_character_encodings (also Cyrillic)
+ */
+class TEscCharsetConverter
+{
+	public const ESC_CHAR_ENCODINGS_MAP = [
+			"\x1B\x25\x47" => 'UTF-8', // ESC-'%G'
```

Comment:
Just wanted to add that these are static and set for the long term.  They haven't changed since they were initially set.  Missing and new should be added on the rarest of times.

There are dates beside the @ todo so we know how long it has been since updated/checked.